notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -23,64 +23,56 @@
 var upmodal_notice = upmodal_notice || {};
 
 if (!upmodal_notice.init) {
-
   upmodal_notice.init = true;
 
-  (function() {
-
+  (function($) {
     var defaults = {
       selectors: {
-        content:         '.np-content',
-        title:           '.np-title',
-        form:            '.np-action-form',
-        body:            '.np-body',
-        link:            '.np-link',
-        actions:         '.np-actions',
-        actionTemplate:  '.np-action-template',
-        closeButton:     '.np-close'
+        content: '.np-content',
+        title: '.np-title',
+        form: '.np-action-form',
+        body: '.np-body',
+        link: '.np-link',
+        actions: '.np-actions',
+        actionTemplate: '.np-action-template',
+        closeButton: '.np-close'
       }
     };
 
     // First 'prime-the-pump' with an ActionURL
-    var initNotices = function($, settings, callback) {
+    var initNotices = function(settings, callback) {
       $.ajax({
-        type:    'POST',
-        url:     settings.invokeNotificationServiceUrl,
-        async:   false,
+        type: 'POST',
+        url: settings.invokeNotificationServiceUrl,
+        async: false,
         success: function() {
-          fetchNotices($, settings, callback);
-        }
+          fetchNotices(settings, callback);
+        },
+        error: console.error
       });
     }
 
     // Then fetch the notifications with a ResourceURL
-    var fetchNotices = function($, settings, callback) {
+    var fetchNotices = function(settings, callback) {
       $.ajax({
-        url:      settings.getNotificationsUrl,
-        type:     'POST',
+        url: settings.getNotificationsUrl,
+        type: 'POST',
         dataType: 'json',
-        success:  function (data) {
-          feed = data.feed;
+        success: function (data) {
+          var feed = data.feed;
           callback(feed);
         },
-        error:    function () {
-          container.html(" ").text("AJAX failed. ~ THE END ~");
-        }
+        error: console.error
       });
     }
 
-    upmodal_notice.launch = function ($, container, options) {
-
+    upmodal_notice.launch = function (container, options) {
```

Comment:
Almost perfect, but there's one part of this refactor I want to alter...

The `launch` method is the one (probably only one) that really _should_ receive a copy of jQuery.  This approach means that the JSP makes the decision about where the copy of jQuery comes from.  That's appropriate b/c the JSP is dynamic, and can implement the choosing in as simple or as complex a way as necessary.  The `.js` file is static, and can't do the same things.